src/pageFunc.js

```diff
@@ -0,0 +1,40 @@
+import Task from './Task.js'
+
+export function enableEdit(){
+  this.parentNode.parentNode.querySelector('.task').disabled = false;
+  this.classList.add('hide');
+  this.parentNode.querySelector('.trashCont').classList.remove('hide');
+}
+
+export function editTask(){
+  const index = parseInt(this.parentNode.querySelector('.task').id,10);
+  this.disabled = true;
+  this.parentNode.querySelector('.icnCont').querySelector('.trashCont').classList.add('hide');
+  this.parentNode.querySelector('.icnCont').querySelector('.editIcon').classList.remove('hide');
+  this.parentNode.querySelector('.task').focus();
+  this.parentNode.querySelector('.task').select();
+  const tasks = JSON.parse(localStorage.getItem('ToDoTasks'));
+  console.log(index);
+  tasks[index].description = this.parentNode.querySelector('.task').value;
+  updateLocalStorage(tasks);
+  document.location.reload();
+}
+
+export function addTask(tasks,desc){
+  const newTask = new Task(desc, tasks.length);
+  tasks.push(newTask);
+  updateLocalStorage(tasks);
+}
+
+export function updateLocalStorage(tasks){
+  localStorage.setItem('ToDoTasks',JSON.stringify(tasks));
+}
```

Comment:
- [ ] This function doesn't work because of the correlation with the task array which the value is incorrect. When adding a new task or refreshing the page, your view is not in sync with the values stored in the LocalStorage. Let's fix that

Reply:
Hello Eric,
This point is not clear to me, this function does work?
Can you please explain your idea?